### PR TITLE
added localization of libdevice and check missing binaries, V207-031

### PR DESCRIPTION
**Pull request by QuentinOchem**

### CODE REVIEW

File: wrapper/src/gnatcuda_wrapper.adb [hunk 1]
```diff
@@ -227,25 +259,55 @@ begin
...
               & gnatvsn.Gnat_Static_Version_String
               & ")");
+            Put_Line ("CUDA Installation: " & CUDA_Root);
 
             Verbose := True;
             LLVM_Arg_Number := @ + 1;
             LLVM_Args (LLVM_Arg_Number) := new String'(Argument (J));
          elsif Arg'Length > 9 and then Arg (1 .. 9) = "-mcpu=sm_" then
             GPU_Name := new String'(Arg (10 .. Arg'Last));
+         elsif Arg'Length > 9 and then Arg (1 .. 17) = "-mcuda-libdevice=" then
```
Comment (leogermond) [inline wrapper/src/gnatcuda_wrapper.adb:274]: Length > 17

Maybe is it a sign that this check needs to be refactored by adding some `function Get_Suffix (S: String; Prefix: String; Suffix : out String_Access) return Boolean`?

File: wrapper/src/gnatcuda_wrapper.adb [hunk 2]
```diff
@@ -205,8 +216,29 @@ function GNATCUDA_Wrapper return Integer is
...
+
+      return Tmp;
+   end Locate_And_Check;
+
    --  Start of processing for GNATCCG_Wrapper
 
+   CUDA_Bin : constant String := GNAT.Directory_Operations.Dir_Name
+     (Locate_And_Check ("ptxas").all);
+   CUDA_Root : constant String := GNAT.Directory_Operations.Dir_Name
+     (CUDA_Bin (CUDA_Bin'First .. CUDA_Bin'Last - 1));
```
Comment (leogermond) [inline wrapper/src/gnatcuda_wrapper.adb:241]: Dir_Name of the file path minus its last character?

**Review by leogermond — CHANGES_REQUESTED** [review]

File: wrapper/src/gnatcuda_wrapper.adb (hunk 1 above)
Comment (QuentinOchem) [reply wrapper/src/gnatcuda_wrapper.adb:274]: Good catch - and good idea. I'll refactor.

File: wrapper/src/gnatcuda_wrapper.adb (hunk 2 above)
Comment (QuentinOchem) [reply wrapper/src/gnatcuda_wrapper.adb:241]: Yes - last character is the directory separator.

Comment (leogermond) [reply wrapper/src/gnatcuda_wrapper.adb:241]: Ah yes I forgot how crusty the lib is.

**Review by leogermond — APPROVED** [review]

**Pull request merged**